lightning/src/ln/peers/handler.rs

```diff
@@ -119,8 +120,68 @@ enum InitSyncTracker{
 	NodesSyncing(PublicKey),
 }
 
+enum PeerState {
+	Authenticating(PeerHandshake),
+	Connected(Conduit),
+}
+
+enum PeerDataProcessingDecision {
+	CompleteHandshake(PublicKey),
+	Continue,
+	Disconnect(PeerHandleError)
+}
+
+impl PeerState {
+	fn is_ready_for_encryption(&self) -> bool {
+		match self {
+			&PeerState::Connected(_) => true,
+			_ => false
+		}
+	}
+
+	fn process_peer_data(&mut self, data: &[u8], mutable_response_buffer: &mut LinkedList<Vec<u8>>) -> PeerDataProcessingDecision {
+		let (new_state_opt, decision) = match self {
+			&mut PeerState::Authenticating(ref mut handshake) => {
+				match handshake.process_act(data) {
+					// Any errors originating from the handshake sequence result in Disconnect
+					Err(_e) => {
+						(None, PeerDataProcessingDecision::Disconnect(PeerHandleError { no_connection_possible: false }))
+					},
+					// Otherwise, handshake may or may not be complete depending on whether or not
+					// we receive a (conduit, pubkey)
+					Ok((response_vec_option, conduit_and_remote_static_public_key_option)) => {
+
+						// Any response generated by the handshake sequence is put into the response buffer
+						if let Some(response_vec) = response_vec_option {
+							mutable_response_buffer.push_back(response_vec);
+						}
+
+						// if process_act() returns the conduit and remote static public key (node id)
+						// the handshake is complete
+						if let Some((conduit, remote_static_public_key)) = conduit_and_remote_static_public_key_option {
+							(Some(PeerState::Connected(conduit)), PeerDataProcessingDecision::CompleteHandshake(remote_static_public_key))
+						} else {
+							(None, PeerDataProcessingDecision::Continue)
+						}
+					}
+				}
+			},
+			&mut PeerState::Connected(ref mut conduit) => {
+				conduit.read(data);
+				(None, PeerDataProcessingDecision::Continue)
+			}
+		};
+
+		if let Some(new_state) = new_state_opt {
+			*self = new_state;
+		}
+
+		decision
+	}
+}
+
 struct Peer {
-	channel_encryptor: PeerChannelEncryptor,
+	encryptor: PeerState,
```

Comment:
No, and the asymmetry between when this was set and when it wasn't was confusing. This has been changed in #692 so that it is an output of the NOISE handshake completion.